nemo/collections/tts/losses/audio_codec_loss.py

```diff
@@ -230,8 +230,8 @@ def output_types(self):
     @typecheck()
     def forward(self, audio_real, audio_gen, audio_len):
         spec_len = (audio_len // self.hop_length) + 1
-        spec_real = self._compute_spectrogram(audio=audio_real, spec_len=spec_len)
-        spec_gen = self._compute_spectrogram(audio=audio_gen, spec_len=spec_len)
+        spec_real = self._compute_spectrogram(audio=audio_real.float(), spec_len=spec_len).to(audio_gen.dtype)
+        spec_gen = self._compute_spectrogram(audio=audio_gen.float(), spec_len=spec_len).to(audio_gen.dtype)
```

Comment:
similar reason as mentioned when using bf16-mixed, autocast to bf16 should be disabled so keeping `audio_real` or `audio_real` with fp32. Otherwise, cast to bf16 and back to fp32 results in loss of precision.